### ISSUE TITLE
[REVIEW]: ordPens: An R package for Selection, Smoothing and Principal Components Analysis for Ordinal Variables

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3828]
**Submitting author:** @ahoshiyar (<a href="http://orcid.org/0000-0002-5702-130X">Aisouda Hoshiyar</a>)
**Repository:** <a href="https://github.com/ahoshiyar/ordPens" target ="_blank">https://github.com/ahoshiyar/ordPens</a>
**Version:** 1.0.1
**Editor:** @osorensen
**Reviewer:** @fartist, @FranjoIM, @mingzehuang
**Archive:** <a href="https://doi.org/10.5281/zenodo.5718572" target="_blank">10.5281/zenodo.5718572</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/2418101c27f855d290a5da28bcee0ea5/status.svg)](https://joss.theoj.org/papers/2418101c27f855d290a5da28bcee0ea5)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/2418101c27f855d290a5da28bcee0ea5"><img src="https://joss.theoj.org/papers/2418101c27f855d290a5da28bcee0ea5/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/2418101c27f855d290a5da28bcee0ea5/status.svg)](https://joss.theoj.org/papers/2418101c27f855d290a5da28bcee0ea5)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/ahoshiyar/ordPens" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@fartist & @FranjoIM & @mingzehuang, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @osorensen know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @fartist
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/ahoshiyar/ordPens">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ahoshiyar/ordPens">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@ahoshiyar) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @FranjoIM
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/ahoshiyar/ordPens">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ahoshiyar/ordPens">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@ahoshiyar) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @mingzehuang
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/ahoshiyar/ordPens">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ahoshiyar/ordPens">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@ahoshiyar) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1310

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1111/j.1467-9876.2010.00753.x is OK
- 10.1214/11-EJS661 is OK
- 10.1111/j.1751-5823.2009.00088.x is OK
- 10.1214/10-aoas355 is OK
- 10.1007/s13253-014-0170-5 is OK
- 10.1515/sagmb-2015-0091 is OK
- 10.31234/osf.io/9qkhj is OK
- 10.1207/s15327906mbr0102_10 is OK
- 10.3390/su11143907 is OK
- 10.1016/s0167-9473(97)00038-8 is OK
- 10.1007/978-981-10-0159-8_2 is OK
- 10.1002/wics.1279 is OK
- 10.1007/bfb0092976 is OK
- 10.1007/bf02591962 is OK
- 10.1007/bf02293871 is OK
- 10.1007/S11336-013-9343-3 is OK
- 10.1007/s11336-018-9623-z is OK
- 10.1177/1471082X16642560 is OK
- 10.4324/9781315128948-36 is OK
- 10.1111/j.1467-9868.2007.00646.x is OK
- 10.1214/ss/1038425655 is OK
- 10.1111/j.1467-9868.2007.00627.x is OK
- 10.1111/j.1467-9868.2005.00490.x is OK
- 10.1111/j.1467-9868.2005.00532.x is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03828/joss.03828/10.21105.joss.03828.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03828/joss.03828/10.21105.joss.03828.pdf) :page_facing_up: :point_left:

**Comment by FranjoIM** [comment]
I am still waiting on @ahoshiyar to address the open issues.

**Comment by fartist** [comment]
I am reviewing it right now. Sent from Mail for Windows From: Franjo IvankovicSent: Wednesday, November 3, 2021 9:37 PMTo: openjournals/joss-reviewsCc: Feeeeeeeeeeng; MentionSubject: Re: [openjournals/joss-reviews] [REVIEW]: ordPens: An R package for Selection, Smoothing and Principal Components Analysis for Ordinal Variables (#3828) I am still waiting on @ahoshiyar to address the open issues.—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by osorensen** [comment]
@fartist @FranjoIM , thanks for responding, and sorry about the automated reminder. I unfortunately cannot turn it off.

**Comment by osorensen** [comment]
@ahoshiyar can you please update us on how it is going addressing the issues raised by the reviewers in the source repository?

**Comment by FranjoIM** [comment]
I have completed the review at this point. The author has addressed my concerns, and pending the review of the other two reviewers, I am recommending this paper for acceptance.

**Comment by osorensen** [comment]
Thanks a lot for your work, @FranjoIM!

**Comment by ahoshiyar** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03828/joss.03828/10.21105.joss.03828.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03828/joss.03828/10.21105.joss.03828.pdf) :page_facing_up: :point_left:

**Comment by ahoshiyar** [comment]
Thank you a lot @FranjoIM !

**Comment by ahoshiyar** [comment]
Dear all, note that the Statement of Need section has been updated (adressing a very interesting comment from @mingzehuang) discussing potential future work/related works.

**Comment by mingzehuang** [comment]
Thank you, @ahoshiyar ! It looks perfectly address my concern! I'm going to finish reviewing the rest of parts ASAP:)

**Comment by osorensen** [comment]
@fartist, could you please update us on how it's going with your review?

**Comment by fartist** [comment]
Please find my review at [https://github.com/ahoshiyar/ordPens/issues/9]([REVIEW]), @ahoshiyar - solid work!

**Comment by osorensen** [comment]
> Thank you, @ahoshiyar ! It looks perfectly address my concern! I'm going to finish reviewing the rest of parts ASAP:)

@mingzehuang, thanks for your great efforts so far! Do you have any other concerns to this submission, or have they been addressed? If so, could you please update your checklist at the top of this page?

**Comment by mingzehuang** [comment]
Great work @ahoshiyar ! My last comment is to generate automated tests which is required by the checklist. [Tests](https://github.com/ahoshiyar/ordPens/issues/10)

**Comment by ahoshiyar** [comment]
@wheden generate pdf

**Comment by ahoshiyar** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03828/joss.03828/10.21105.joss.03828.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03828/joss.03828/10.21105.joss.03828.pdf) :page_facing_up: :point_left:

**Comment by mingzehuang** [comment]
I have completed the review. The author has perfectly addressed my concerns, I am recommending this paper for acceptance:)

**Comment by osorensen** [comment]
@fartist, there is one unchecked box in your review checklisit at the top of this thread, regarding references. Could you please check if this issue has now been resolved?

**Comment by osorensen** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03828/joss.03828/10.21105.joss.03828.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03828/joss.03828/10.21105.joss.03828.pdf) :page_facing_up: :point_left:

**Comment by fartist** [comment]
Thanks - @osorensen,  have completed the review. 
I have no further comments - @ahoshiyar did a great job addressing my concerns - I recommend it to be accepted.

**Comment by osorensen** [comment]
Thanks a lot to @fartist, @FranjoIM, and @mingzehuang for their reviews. You work is very much appreciated!

@ahoshiyar, I will now read through the paper once again, and get back to you if I have any furter comments. In the meantime, could you please do the following:

- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

**Comment by ahoshiyar** [comment]
Thank you so much for your comments, @osorensen! 
The tagged version is 1.0.1 & Zenodo DOI is 10.5281/zenodo.5718572

**Comment by ahoshiyar** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03828/joss.03828/10.21105.joss.03828.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03828/joss.03828/10.21105.joss.03828.pdf) :page_facing_up: :point_left:

**Comment by osorensen** [comment]
@whedon set 10.5281/zenodo.5718572 as arcive

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by osorensen** [comment]
@whedon set 10.5281/zenodo.5718572 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5718572" target="_blank">10.5281/zenodo.5718572</a> is the archive.

**Comment by osorensen** [comment]
@whedon set 1.0.1 as version

**Comment by whedon** [comment]
OK. 1.0.1 is the version.

**Comment by osorensen** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1111/j.1467-9876.2010.00753.x is OK
- 10.1214/11-EJS661 is OK
- 10.1111/j.1751-5823.2009.00088.x is OK
- 10.1214/10-aoas355 is OK
- 10.1007/s13253-014-0170-5 is OK
- 10.1515/sagmb-2015-0091 is OK
- 10.31234/osf.io/9qkhj is OK
- 10.1207/s15327906mbr0102_10 is OK
- 10.3390/su11143907 is OK
- 10.1016/s0167-9473(97)00038-8 is OK
- 10.1007/978-981-10-0159-8_2 is OK
- 10.1002/wics.1279 is OK
- 10.1007/bfb0092976 is OK
- 10.1007/bf02591962 is OK
- 10.1007/bf02293871 is OK
- 10.1007/S11336-013-9343-3 is OK
- 10.1007/s11336-018-9623-z is OK
- 10.1177/1471082X16642560 is OK
- 10.4324/9781315128948-36 is OK
- 10.1111/j.1467-9868.2007.00646.x is OK
- 10.1214/ss/1038425655 is OK
- 10.1111/j.1467-9868.2007.00627.x is OK
- 10.1111/j.1467-9868.2005.00490.x is OK
- 10.1111/j.1467-9868.2005.00532.x is OK
- 10.21105/joss.03634 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2762

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2762, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@ahoshiyar I have proofread your paper which seems in order. However the ZENODO archive needs changes as listed below:

- [ ] Please amend the license on the ZENODO archive (you have to do this manually) to match your software license: https://github.com/ahoshiyar/ordPens/blob/master/LICENSE, it currently lists cc-by.

**Comment by ahoshiyar** [comment]
Thank you @Kevin-Mattheus-Moerman, I see.. 
However, Zenodo seems to provide only the Creative Commons Attribution licenses. I've found also other JOSS related packages having different repo licenses and Zenodo licenses and was confused. 
Could you please give me advice on what to do in this case?

**Comment by Kevin-Mattheus-Moerman** [comment]
@arfon can you comment on this. Shouldn't the zenodo archive be editable to change the license to match that of the software?

**Comment by arfon** [comment]
I think this may be a limitation of how Zenodo works. As long as the software has a license file in the repository (which it will) with an OSI approved license then I think this is ok

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1467794726250332164) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2795
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03828](https://doi.org/10.21105/joss.03828)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by Kevin-Mattheus-Moerman** [comment]
@ahoshiyar congratulations on getting this work published in JOSS! 

Thanks @fartist, @FranjoIM, @mingzehuang for you review efforts, and thank you @osorensen for editing this work!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03828/status.svg)](https://doi.org/10.21105/joss.03828)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03828">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03828/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03828/status.svg
   :target: https://doi.org/10.21105/joss.03828
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03828/status.svg)](https://doi.org/10.21105/joss.03828)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
